src/main/java/io/github/moulberry/notenoughupdates/listener/ItemTooltipEssenceShopListener.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2022 NotEnoughUpdates contributors
+ *
+ * This file is part of NotEnoughUpdates.
+ *
+ * NotEnoughUpdates is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * NotEnoughUpdates is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with NotEnoughUpdates. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.moulberry.notenoughupdates.listener;
+
+import com.google.gson.JsonObject;
+import io.github.moulberry.notenoughupdates.NotEnoughUpdates;
+import io.github.moulberry.notenoughupdates.core.util.StringUtils;
+import io.github.moulberry.notenoughupdates.util.Utils;
+import net.minecraftforge.event.entity.player.ItemTooltipEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ItemTooltipEssenceShopListener {
+	private final NotEnoughUpdates neu;
+
+	public ItemTooltipEssenceShopListener(NotEnoughUpdates neu) {
+		this.neu = neu;
+	}
+
+	@SubscribeEvent
+	public void slayerRngChance(ItemTooltipEvent event) {
+		if (!neu.isOnSkyblock()) return;
+		if (event.toolTip == null) return;
+		if (!Utils.getOpenChestName().endsWith(" Essence Shop")) return;
+		if (!NotEnoughUpdates.INSTANCE.config.tooltipTweaks.essencePriceInEssenceShop) return;
+
+		List<String> newToolTip = new ArrayList<>();
+		boolean next = false;
+		for (String line : event.toolTip) {
+
+			if (next) {
+				next = false;
+				String string = StringUtils.cleanColour(line);
+				String[] split = string.split(" ");
+				String rawNumber = split[0].replace(",", "");
+				int amount = Integer.parseInt(rawNumber);
+				String type = split[1];
```

Comment:
Can we just have a try catch for all of these operations, in case hypixel decides to change the format of this, (or the user has another mod messing with tooltips)